resources/lemmas.md

```diff
@@ -85,6 +86,21 @@ The following lemmas essentially capture the signature extraction mechanisms.
 It reduces the reasoning efforts of the underlying theorem prover, factoring out the essence of the byte-twiddling operations.
 
 ```k
+    syntax Bool ::= #isRegularWordStack ( WordStack ) [function]
+ // -------------------------------------------------------
+    rule #isRegularWordStack(N : WS => WS)
+    rule #isRegularWordStack(.WordStack) => true
+
+    //Rules for #padToWidth with regular symbolic arguments.
+    //Same as for concrete #padToWidth, when WordStack is of regular form "A:B ... :.WordStack"
+    //Not clear why KEVM rules for #padToWidth were marked [concrete]. If they were general, rules below would not be necessary.
```

Comment:
hmm,, good point.  the reason why i added [concrete] attribute is to ensure that the rules are deterministic even in the presence of the lemma rules in lemmas.md.  but it may not be necessary.

why don't you try dropping [concrete] attribute there, and see if the tests pass? (you can do that in a separate PR though.)

Reply:
will do that in another PR.